jetbrains-core/tst/software/aws/toolkits/jetbrains/services/cloudwatch/logs/CloudWatchLogStreamCoroutineTest.kt

```diff
@@ -0,0 +1,182 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.cloudwatch.logs
+
+import com.intellij.testFramework.ProjectRule
+import com.intellij.ui.table.TableView
+import com.intellij.util.ui.ListTableModel
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.channels.ClosedSendChannelException
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.isActive
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.test.TestCoroutineScope
+import kotlinx.coroutines.withTimeout
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.assertThatThrownBy
+import org.junit.After
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.Mockito
+import org.mockito.Mockito.`when`
+import software.amazon.awssdk.services.cloudwatchlogs.CloudWatchLogsAsyncClient
+import software.amazon.awssdk.services.cloudwatchlogs.model.GetLogEventsRequest
+import software.amazon.awssdk.services.cloudwatchlogs.model.GetLogEventsResponse
+import software.amazon.awssdk.services.cloudwatchlogs.model.OutputLogEvent
+import software.aws.toolkits.jetbrains.core.MockClientManagerRule
+import software.aws.toolkits.resources.message
+import java.util.concurrent.CompletableFuture
+
+// ExperimentalCoroutinesApi is needed for TestCoroutineScope
+@ExperimentalCoroutinesApi
+class CloudWatchLogStreamCoroutineTest {
+    @JvmField
+    @Rule
+    val projectRule = ProjectRule()
+
+    @JvmField
+    @Rule
+    val mockClientManagerRule = MockClientManagerRule(projectRule)
+
+    private val testCoroutineScope: TestCoroutineScope = TestCoroutineScope()
+
+    @After
+    fun after() {
+        testCoroutineScope.cleanupTestCoroutines()
+    }
+
+    @Test
+    fun modelIsPopulated() {
+        val mockClient = mockClientManagerRule.create<CloudWatchLogsAsyncClient>()
+        `when`(mockClient.getLogEvents(Mockito.any<GetLogEventsRequest>()))
```

Comment:
if we're still using mockito-kotlin, should be `whenever` instead of `` `when` ``